### ISSUE TITLE
[bug] Can't use responsive `space`

### ISSUE THREAD

**Issue by nandorojo** [tamagui/tamagui#304]
**Describe the bug**

This always uses `$4`:

```tsx
<View 
  space="$4"
  $md={{
    space: '$5',
  }}
/>
```

Once we have flex gap in RN 0.71 it should get fixed. That said, I think the best solution to this would be to allow responsiveness per-style property, like Dripsy and Stitches. I find it much better to co-locate responsiveness per-property:

```tsx
<View 
  space={{
    $base: '$4',
    $md: '$5'
  }}
/>
```

This API would also presumably this issue, since the `space` gets passed down to the spacer element to size it.

Lmk if a repro etc is necessary. Seemed pretty straightforward.

**Comment by natew** [comment]
Yea this is a gap (heh) in functionality right now, though I think I can use the new `useMediaPropsActive` and fix it.

**Comment by natew** [comment]
<img width="526" alt="image" src="https://user-images.githubusercontent.com/12100/198855431-f96981fd-ca38-4f56-ab8e-15922bebe048.png">

Seems I do have the logic for this set up so there must be some regression here, testing.

**Comment by natew** [comment]
Fixed in 9d77105de877725e78ddebfc297b846b83f3e53f
